geode-core/src/distributedTest/java/org/apache/geode/security/SecurityManagerAvailabilityDUnitTest.java

```diff
@@ -0,0 +1,164 @@
+      InternalCache cache = ClusterStartupRule.getCache();
+      TestIntegratedSecurityService securityService =
+          (TestIntegratedSecurityService) cache.getSecurityService();
+      securityService.setGetSubjectFailConditions(50, UnavailableSecurityManagerException.class);
+      cache.createRegionFactory(RegionShortcut.REPLICATE).create("region");
+    }, serverVM0, serverVM1);
+
+    int svm0 = serverVM0.getPort();
+    int svm1 = serverVM1.getPort();
+    clientVM = clusterStartupRule.startClientVM(3,
+        c -> c.withServerConnection(svm0, svm1)
+            .withPoolSubscription(true)
+            .withProperty(SECURITY_CLIENT_AUTH_INIT, UpdatableUserAuthInitialize.class.getName()));
+  }
+
+  @Test
+  public void confirmSecurityManagerAvailabilityExceptionDoesNotStopClientPutOperations() {
+    Integer tries = 1000;
+    List<Object> clientResult = clientVM.invoke(() -> {
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      UpdatableUserAuthInitialize.setUser("data1");
+      Region<Object, Object> region =
+          clientCache.createClientRegionFactory(ClientRegionShortcut.PROXY).create("region");
+
+      List<Object> returns = new ArrayList<>(2);
+      returns.add(new Exception("no exception"));
+      ZE_COUNT.set(0);
+      while (ZE_COUNT.get() < tries) {
+        try {
+          int key = ZE_COUNT.incrementAndGet();
+          region.put(key, "value" + key);
+        } catch (Throwable e) {
+          Throwable cause = e.getCause();
+          Throwable causeCause = cause != null ? cause.getCause() : null;
+          if (cause instanceof UnavailableSecurityManagerException
+              || causeCause instanceof UnavailableSecurityManagerException) {
+            returns.set(0, e);
+            break;
+          }
+        }
+      }
+      returns.add(ZE_COUNT.get());
+      return returns;
+    });
+
+    Map<String, Object> resultMap0 = serverVM0.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      TestIntegratedSecurityService securityService =
+          (TestIntegratedSecurityService) cache.getSecurityService();
+      return securityService.getGetSubjectFailInformation();
+    });
+
+    Map<String, Object> resultMap1 = serverVM1.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      TestIntegratedSecurityService securityService =
+          (TestIntegratedSecurityService) cache.getSecurityService();
+      return securityService.getGetSubjectFailInformation();
+    });
+
+    assertThat((Integer) resultMap0.get(FAIL_TIMES)).isGreaterThan(0);
+    assertThat((String) resultMap0.get(FAIL_THROWABLE)).isNotEmpty();
+    assertThat((Integer) resultMap1.get(FAIL_TIMES)).isGreaterThan(0);
+    assertThat((String) resultMap1.get(FAIL_THROWABLE)).isNotEmpty();
+
+    Throwable error = (Throwable) clientResult.get(0);
+
+    assertThat(error instanceof Exception).isTrue();
+    assertThat(error.getMessage()).isEqualTo("no exception");
```

Comment:
These might be better as:
```
    assertThat(error).isInstanceOf(Exception.class);
    assertThat(error).hasMessage("no exception");
```
Also, I think that the first assertion is always true, since the exception can either be the `Exception` set on line 119 or some other exception with `UnavailableSecurityManagerException` as a cause, set on line 130, so this assertion isn't really telling us much. When I reverted the fix and reran the test, this assertion did not fail (although the one on the message contents did).